Sources/ScriptingCore/utils.swift

```diff
@@ -0,0 +1,121 @@
+            return nil
+        }
+        absolutePath = AbsolutePath(cwd, path)
+    }
+    guard localFileSystem.isFile(absolutePath) else {
+        return nil
+    }
+    return absolutePath
+}
+
+public func checkAndPerformCache(for file: String, at dirPath: AbsolutePath) throws -> (productName: String, cacheDirPath: AbsolutePath) {
+    if let scriptPath = resolveFilePath(file) {
+        let json = try ScriptParse.manifest(for: scriptPath)
+        let decoder = JSONDecoder()
+        let manifest = try decoder.decode(ScriptDependencies.self, from: json)
+        
+        let productName = scriptPath.basename.spm_dropSuffix(".swift").spm_mangledToBundleIdentifier()
+        let cacheDirPath = dirPath.appending(component: "\(productName)-\(scriptPath.sha256Hash)")
+        try localFileSystem.createDirectory(cacheDirPath, recursive: true)
+        
+        let sourceDirPath = cacheDirPath.appending(components: "Sources", productName)
+        try localFileSystem.createDirectory(sourceDirPath, recursive: true)
+        if !localFileSystem.exists(sourceDirPath.appending(component: "main.swift")) {
+            try localFileSystem.createSymbolicLink(sourceDirPath.appending(component: "main.swift"), pointingAt: scriptPath, relative: false)
+        }
+        
+        var targets: [(name: String, package: String)] = []
+        for package in manifest.modules {
+            let packageName = package.package.name
+            for target in package.modules {
+                targets.append((target, packageName))
+            }
+        }
+        
+        let packageSwift = """
+            // swift-tools-version:5.4
+            import PackageDescription
+            let package = Package(
+                name: "\(productName)",
+                products: [
+                    .executable(
+                        name: "\(productName)",
+                        targets: ["\(productName)"]),
+                ],
+                dependencies: [\(manifest.modules.map{".package(\($0.package.raw))"}
+                        .joined(separator: ", "))],
+                targets: [
+                    .executableTarget(
+                        name: "\(productName)",
+                        dependencies: [\(targets.map {".product\($0)"}.joined(separator: ", "))],
+                        swiftSettings: [.unsafeFlags(["-Xfrontend", "-ignore-package-declarations"])]
+                )]
+            )
+            
+            """
+        
+        try localFileSystem.writeIfChanged(
+            path: cacheDirPath.appending(component: "Package.swift"),
+            bytes: ByteString(stringLiteral: packageSwift)
+        )
+        return (productName, cacheDirPath)
+    } else {
+        let cacheDirPath = dirPath.appending(component: file)
+        guard localFileSystem.isDirectory(cacheDirPath) else {
+            throw ScriptError.fileNotFound(file)
+        }
+        return (String(file.dropLast(7)), cacheDirPath)
```

Comment:
This hardcoded 7 seems a little shady — what is the suffix that is being dropped here?  Is it "Sources"?  There is probably a clearer / safer way to do this.  I think there's a helper function to drop a suffix if it exists on the path.

Reply:
It's not a hardcoded one. These should be hyphen and the six hash digits.
```swift
let cacheDirPath = dirPath.appending(component: "\(productName)-\(scriptPath.sha256Hash)")
```
Since these characters are all appended in this file, I think it's okay to hardcode? Or maybe split the name with `-` and emit the last part?